runtime/bcverify/rtverify.c

```diff
@@ -2615,7 +2612,7 @@ j9rtv_verifyArguments (J9BytecodeVerificationData *verifyData, J9UTF8 * utf8stri
 			}
 
 			baseType = (UDATA) argTypeCharConversion[*signature - 'A'];
-			/* Jazz 82615: Tagged as 'bool' when argTypeCharConversion returns BCV_BASE_TYPE_BYTE_BIT for type 'Z' */
+			/* Jazz 82615: Tagged as 'bool' when argTypeCharConversion returns BCV_BASE_TYPE_INT for type 'Z' */
 			if ('Z' == *signature) {
```

Comment:
Can this be removed?  The copy of this above in the same method was deleted.

Reply:
The logic here is specific to base types which is different from above (base type array) and they are using different conversion tables against the VM Spec.

1) For base type array, type [B and type [Z are considered as different. To follow the Spec, we already changed the conversion table (baseTypeCharConversion) to support type [Z as follows:
const U_32 baseTypeCharConversion[] = {
... ...
0,						BCV_BASE_TYPE_BOOL_BIT (type [Z),		0};
So there is no need to keep if ('Z' == *signature) {...} there as baseType is set to BCV_BASE_TYPE_BOOL_BIT for type [Z, which is expected in generating the error messages.

2) For base type, the VM Spec says at  4.10.1.2 Verification Type System:
The primitive types byte, char, short, and boolean (field descriptors B, C, S, Z) all correspond to the verification type int. In such case, the conversion table argTypeCharConversion must remain unchanged.

const U_32 argTypeCharConversion[] = {
... ...
0,					BCV_BASE_TYPE_INT (type Z),		0}; 

As a consequence, baseType is set to BCV_BASE_TYPE_INT for type Z.
So, "if ('Z' == *signature) {...}" needs to be here to ensure we can print out base type Z in the error messages.

I already updated the comments to explain why we need to keep the if statement.